src/main/java/com/microsoft/azure/spring/data/cosmosdb/core/query/DocumentQuery.java

```diff
@@ -55,7 +55,12 @@ private boolean isCrossPartitionQuery(@NonNull String keyName) {
 
         final Optional<Criteria> criteria = this.getSubjectCriteria(this.criteria, keyName);
 
-        return criteria.map(criteria1 -> criteria1.getType() == CriteriaType.IS_EQUAL).orElse(true);
+        return criteria.map(criteria1 -> criteria1.getType() != CriteriaType.IS_EQUAL).orElse(true);
+    }
+
+    private boolean hasKeywordOr() {
+        // If there is OR keyword in DocumentQuery, the top node of Criteria must be OR type.
+        return this.criteria.getType() == CriteriaType.OR;
```

Comment:
Why not need to check the `subCriteria` inside the `criteria` ?

Reply:
Actually, here we check is there any OR keyword in criteria. Any OR keyword will make the semantic to query cross ParititonKey.